drivers/display/display_st7735r.c

```diff
@@ -268,7 +269,7 @@ static void st7735r_get_capabilities(const struct device *dev,
 	memset(capabilities, 0, sizeof(struct display_capabilities));
 	capabilities->x_resolution = config->width;
 	capabilities->y_resolution = config->height;
-	if (config->madctl & ST7735R_MADCTL_BGR) {
+	if ((bool)(config->madctl & ST7735R_MADCTL_BGR) ^ config->swap_color_order) {
```

Comment:
Why is there cast to `(bool)`?

Reply:
XOR is bitwise operator, it cannot calculate correctly without cast.
It was not good notation to understanding.
Rewrited it with NOT operator.